packages/astro/src/runtime/client/dev-overlay/overlay.ts

```diff
@@ -20,6 +23,7 @@ export class AstroDevOverlay extends HTMLElement {
 	plugins: DevOverlayPlugin[] = [];
 	HOVER_DELAY = 750;
 	hasBeenInitialized = false;
+	customPluginsToShow = 3;
```

Comment:
Currently, this is hardcoded. In the future, we plan to make it so it depends on the width of your screen. Unfortunately, this requires quite some refactoring in how the dev overlay UI is built to ensure compatibility with view transitions and stuff, so will tackle it later.